llvm/tools/llvm-extract/llvm-extract.cpp

```diff
@@ -355,9 +358,25 @@ int main(int argc, char **argv) {
       for (StringRef BBName : P.second) {
         // The function has been materialized, so add its matching basic blocks
         // to the block extractor list, or fail if a name is not found.
+#ifndef NDEBUG
         auto Res = llvm::find_if(*P.first, [&](const BasicBlock &BB) {
-          return BB.getName() == BBName;
+          return BB.getNameOrAsOperand() == BBName;
```

Comment:
we should make `getNameOrAsOperand()` available in non-debug builds as well to avoid the duplicated logic here